File: game-server/src/main/java/work/fking/pangya/game/packet/outbound/LoginBonusClaimReplyPacket.java
package work.fking.pangya.game.packet.outbound;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.networking.protocol.OutboundPacket;

public class LoginBonusClaimReplyPacket implements OutboundPacket {

    private static final int ID = 0x0249;

    @Override
    public void encode(ByteBuf target) {
        target.writeShortLE(ID);

        target.writeZero(5); // padding
        target.writeIntLE(402653190); // item iffId
        target.writeIntLE(55); // quantity

        target.writeIntLE(402653190); // future item iffId
        target.writeIntLE(40); // future quantity

        target.writeIntLE(22); // current day streak
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/packet/outbound/LoginBonusClaimReplyPacket.kt
package work.fking.pangya.game.packet.outbound

import io.netty.buffer.ByteBuf
import work.fking.pangya.networking.protocol.OutboundPacket

class LoginBonusClaimReplyPacket : OutboundPacket {

    override fun encode(buffer: ByteBuf) {
        buffer.writeShortLE(0x249)
        buffer.writeZero(5) // padding
        buffer.writeIntLE(402653190) // item iffId
        buffer.writeIntLE(55) // quantity
        buffer.writeIntLE(402653190) // future item iffId
        buffer.writeIntLE(40) // future quantity
        buffer.writeIntLE(22) // current day streak
    }
}
